src/test/kotlin/com/ampnet/web3provider/controller/ProviderControllerTest.kt

```diff
@@ -1,68 +1,37 @@
 import org.mockito.Mockito
-import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.boot.test.context.SpringBootTest
-import org.springframework.boot.test.mock.mockito.MockBean
 import org.springframework.http.MediaType
-import org.springframework.restdocs.RestDocumentationContextProvider
-import org.springframework.restdocs.RestDocumentationExtension
-import org.springframework.restdocs.mockmvc.MockMvcRestDocumentation
-import org.springframework.restdocs.operation.preprocess.Preprocessors
-import org.springframework.test.context.junit.jupiter.SpringExtension
-import org.springframework.test.web.servlet.MockMvc
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
-import org.springframework.test.web.servlet.setup.DefaultMockMvcBuilder
-import org.springframework.test.web.servlet.setup.MockMvcBuilders
-import org.springframework.web.context.WebApplicationContext
 import org.web3j.protocol.core.DefaultBlockParameterName
 import org.web3j.utils.Numeric
 import java.math.BigInteger
 
-@ExtendWith(value = [SpringExtension::class, RestDocumentationExtension::class])
-@SpringBootTest
-class ProviderControllerTest : TestBase() {
+class ProviderControllerTest : ControllerTestBase() {
 
     private val address = "0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B"
     private val blockParameter = DefaultBlockParameterName.LATEST.value
-    private val balance = Numeric.encodeQuantity(BigInteger.TEN)
-    private val code = Numeric.encodeQuantity(BigInteger.ONE)
+    private val hexStringResult = Numeric.encodeQuantity(BigInteger.TEN)
+    private val txHash = "0x88df016429689c079f3b2f6ad39fa052532c56795b733da78a91ebe6a713944b"
 
     private lateinit var testContext: TestContext
 
-    @Autowired
-    private lateinit var objectMapper: ObjectMapper
-
-    private lateinit var mockMvc: MockMvc
-
-    @MockBean
-    private lateinit var defaultProviderService: DefaultProviderService
-
     @BeforeEach
-    fun init(wac: WebApplicationContext, restDocumentation: RestDocumentationContextProvider) {
-        mockMvc = MockMvcBuilders.webAppContextSetup(wac)
-            .apply<DefaultMockMvcBuilder>(MockMvcRestDocumentation.documentationConfiguration(restDocumentation))
-            .alwaysDo<DefaultMockMvcBuilder>(
-                MockMvcRestDocumentation.document(
-                    "{ClassName}/{methodName}",
-                    Preprocessors.preprocessRequest(Preprocessors.prettyPrint()),
-                    Preprocessors.preprocessResponse(Preprocessors.prettyPrint())
-                )
-            )
-            .build()
+    fun init() {
         testContext = TestContext()
+        cacheCleanerService.deleteBalance(address + blockParameter)
+        cacheCleanerService.deleteCode(address + blockParameter)
+        cacheCleanerService.deleteChainId()
+        cacheCleanerService.deleteTransactionByHash(txHash)
```

Comment:
You can create only one function to clean the cache because you will want an empty cache for each test case. There is no needed to clean cache by key, just dump all values.